core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java

```diff
@@ -295,14 +294,14 @@ public void serverStarting(Server server) throws Exception {
             @Override
             public void serverStarted(Server server) {
                 if (startHealthy) {
-                    serverHealth.setHealthy(true);
+                    serverHealth.setHealthStatus(HealthStatus.HEALTHY);
                 }
             }
 
             @Override
             public void serverStopping(Server server) {
                 serverStopping = true;
-                serverHealth.setHealthy(false);
+                serverHealth.setHealthStatus(HealthStatus.UNHEALTHY);
```

Comment:
Should we also introduce `HealthStatus.UNDER_MAINTENANCE`?

Reply:
For finer grained control, it seems a great idea. 

What do you think of the below responses for each status?(I'm a bit worried if this will break backward compatibility though)  
- HEALTHY: `{"healthy":true, "status": "HEALTHY"}` 
- DEGRADED: {"healthy":true, "status": "DEGRADED"}` 
- UNDER_MAINTENANCE: {"healthy":true, "status": "UNDER_MAINTENANCE"}`  
- UNHEALTHY: {"healthy":false, "status": "UNHEALTHY"}`

Reply:
Sure, I'll add it. 👍